### PR TITLE
Query : Get Ancestors

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose
Adds Support for Get Ancestors

## Changes
Adds Support for Get Ancestors

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance
By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 1]
```diff
@@ -325,6 +326,29 @@ public Semver getVersion() {
...
     }
 
+    /**
+     * Gets Block Ancestor Blocks.
+     * @param blockHash {@link Hash} of the block.
+     * @param num Total no of Ancestor blocks to get.
+     * @return {@link BlockAncestors}
+     * @throws BlockNotFoundException When the returned response from Node is invalid or null.
+     */
+    public BlockAncestors getAncestors(Hash blockHash, long num) throws BlockNotFoundException {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:336]: I think this should just return `ImmutableList<Hash>`.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/getancestors/BlockAncestors.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+@Builder
+@Getter
+/**
+ * Represents Ancestors of a Block.
+ */
+public class BlockAncestors {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/getancestors/BlockAncestors.java:20]: I don't think this abstraction provides much functionality, so I propose to simply delete it and return `ImmutableList<Hash>`.

File: README.md [hunk 3]
```diff
@@ -220,6 +220,12 @@ Retrives the node software version.
 NodeInfo getNodeInfo()
 ```
 
+- `getAncestors`
+```java
+BlockAncestors getAncestors(Hash blockHash, long num) throws BlockNotFoundException
+```
+Gets Ancestor Blocks for the provided input Block Hash.
```
Comment (EmilLa1) [inline README.md:227]:
```suggestion
Get a list of block hashes that preceding the provided block hash and with a maximum size of the provided number.
```

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Looks generally good, I don't think the `BlockAncestors` type brings much value to the table right now, so I suggest that we simply return `ImmutableList<Hash>`.

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Even it's a simple query, I urge you to add a unit test for parsing the returned JSON.
To do that you probably want to factor out the deserialization logic into a separate helper function. 

Otherwise looks good.

**Comment by parv0888** [comment]
> Even it's a simple query, I urge you to add a unit test for parsing the returned JSON. To do that you probably want to factor out the deserialization logic into a separate helper function.
> 
> Otherwise looks good.

Thanks, Added unit tests

**Review by EmilLa1 — APPROVED** [review]
Looks good

**Pull request merged**